src/ai/backend/agent/stats.py

```diff
@@ -555,6 +574,35 @@ async def collect_node_stat(self) -> None:
             agent_id, serialized_agent_updates, expire_sec=self.cache_lifespan
         )
 
+    def _extract_slot_measurement_scale_factor(
+        self,
+        slot_names: Sequence[SlotName],
+        metric_key: MetricKey,
+    ) -> Decimal:
+        match slot_names:
+            case []:
+                return Decimal(1)
+            case [slot_name]:
+                return self.agent.resource_partitioner.get_resource_scaling_factor(slot_name)
+            case _:
+                log.warning(
+                    "Plugin defines more than 1 device slot info. Attempting best-effort match with metric key..."
+                )
+                slot_names_with_same_prefix_as_metric_key = [
+                    slot_name
+                    for slot_name in slot_names
+                    if metric_key.startswith(slot_name.split(".")[0])
+                ]
+                if len(slot_names_with_same_prefix_as_metric_key) == 1:
+                    slot_name = slot_names_with_same_prefix_as_metric_key[0]
+                    log.info(f"Found slot name {slot_name} with same prefix as {metric_key}")
+                    return self.agent.resource_partitioner.get_resource_scaling_factor(slot_name)
+                else:
+                    raise ValueError(
+                        f"Plugin defines more than 1 device slots {slot_names}, "
+                        f"and matching with metric key {metric_key} yielded no results."
+                    )
```

Comment:
Exposing the resource partitioner externally doesn’t seem ideal. 
Unlike a simple dataclass that only holds values, for classes that perform specific actions, it’s better not to expose internal fields externally. Doing so creates coupling with external code, which makes future modifications difficult.